### PR TITLE
Enable torch.compile for loss_fn

### PR DESCRIPTION
**Pull request by BoyuanFeng**
This PR enables `torch.compile` for loss functions, which mitigates the performance gap between PyTorch and JAX for fastmri (11.6% -> 1.83%), librispeech_deepspeech (12.02% -> -15.01%), ogbg (6.06% -> 3.28%), and wmt (19.75% -> 3.52%).

Close: #467 
Close: #488

### CODE REVIEW

**Review by priyakasimbeg — CHANGES_REQUESTED** [review]
Thanks for getting this PR in!
It seems like there is a small linting error from one of the lines in submission_runner.py. Could you fix this please? If you have yapf==0.32 installed you can just run `yapf -i -r -vv -p submission_runner.py` and commit the changes.

**Review by msaroufim — APPROVED** [review]

**Review by priyakasimbeg — APPROVED** [review]
